deluge/ui/ui_entry.py

```diff
@@ -32,65 +34,92 @@ def start_ui():
     """Entry point for ui script"""
     lang.setup_translations()
 
-    # Setup the argument parser
-    parser = BaseArgParser()
-    group = parser.add_argument_group(_("UI Options"))
-
+    # Get the registered UI entry points
     ui_entrypoints = dict([(entrypoint.name, entrypoint.load())
                            for entrypoint in pkg_resources.iter_entry_points("deluge.ui")])
+    ui_titles = sorted(ui_entrypoints.keys())
 
-    cmd_help = [_("The UI that you wish to launch. The UI choices are:")]
-    cmd_help.extend(["%s -- %s" % (k, getattr(v, "cmd_description", "")) for k, v in ui_entrypoints.iteritems()])
-
-    group.add_argument("-u", "--ui", action="store",
-                       choices=ui_entrypoints.keys(), help="\n  * ".join(cmd_help))
-    group.add_argument("-a", "--args", action="store",
-                       help=_('Arguments to pass to the UI. Multiple args must be quoted, e.g. -a "--option args"'))
-    group.add_argument("-s", "--set-default-ui", dest="default_ui", choices=ui_entrypoints.keys(),
-                       help=_("Sets the default UI to be run when no UI is specified"), action="store")
+    # Setup the initial argument parser for the UI selection
+    parser = BaseArgParser()
+    group = parser.add_argument_group(_("UI Options"))
+    group.add_argument("-s", "--set-default-ui", dest="default_ui", choices=ui_titles,
+                       help=_("Sets the default UI to be run when no UI is specified"))
 
-    options = parser.parse_args(deluge.common.unicode_argv()[1:])
+    dubious_args = ["-h", "--help", "-v", "-V", "--version"]
+    # Parse arguments without help to avoid it being triggered on intial parsing
+    args = [a for a in sys.argv if a not in dubious_args]
+    options = parser.parse_known_ui_args(args)
 
     config = deluge.configmanager.ConfigManager("ui.conf", DEFAULT_PREFS)
     log = logging.getLogger(__name__)
+    log.info("Deluge ui %s", deluge.common.get_version())
 
     if options.default_ui:
         config["default_ui"] = options.default_ui
         config.save()
         log.info("The default UI has been changed to %s", options.default_ui)
         sys.exit(0)
 
-    log.info("Deluge ui %s", deluge.common.get_version())
-    log.debug("options: %s", options)
-
-    selected_ui = options.ui if options.ui else config["default_ui"]
-
-    config.save()
+    default_ui = config["default_ui"]
+    config.save()  # Save in case config didn't already exist
     del config
 
-    # reconstruct arguments to hand off to child client
-    client_args = []
-    if options.args:
-        import shlex
-        client_args.extend(shlex.split(options.args))
+    # We have parsed and got the config dir needed to get the default UI
+    # Now create a parser for choosing the UI
+    parser = BaseArgParser(common_help=True)
+    group = parser.add_argument_group(_("UI Options"))
```

Comment:
I think you should replace these 3 lines with:

```
# Reuse UI Options group to actually display the group text to the user. The result from these args is not used.
parser._action_groups.append(group)
```

I know it's using a private method but the duplication is hurting my eyes :P